app/PropositionFactory.php

```diff
@@ -67,6 +71,9 @@ public function getFlagCount($id) {
 	public function getFlagTypeCount($id, $type) {
 		return Flags::where('proposition', [$id])->where('type', $type)->count();
 	}
+	public function getGlobalFlagCount() {
+		return Flags::count();
```

Comment:
Yes! You can remove `groupBy` and `join` as you are only counting flags and don't need any information about propositions